### ISSUE TITLE
Feature Request: ability to ignore host validation while testing

### ISSUE THREAD

**Issue by danhammari** [celtic-project/LTI-PHP#26]
In my test environment I've spun up dev instances of my servers and given them self-signed certificates for https. When I initiate LTI 1.3 messages between my platform and tool I get the following response:

```
["error"]=>
  string(48) "SSL certificate problem: self signed certificate"
```

I can bypass this error message if I add the following lines to the `ceLTIc\LTI\Http\CurlClient` class:
```
        if( $test_environment ){
            curl_setopt( $ch, CURLOPT_SSL_VERIFYHOST, false );
            curl_setopt( $ch, CURLOPT_SSL_VERIFYPEER, false );
        }
```

I would need a way to set the `$test_environment` boolean, perhaps by passing it into the `send()` method as an optional parameter.

Have you run into this issue before, and do you have a different means of bypassing the self-signed certificate check? If not, do you think it would be worthwhile to add a testing flag to the CurlClient's `send()` method? I am most interested in enabling PHP unit tests and think this could be helpful.

**Comment by spvickers** [comment]
Thanks for the message; I have not seen this suggestion before.  The possible solutions which come to mind are:
- use http rather than https;
- adapt the CurlClient class to your own needs and set this to be used with the Http\HttpMessage::setHttpClient method.
Hopefully one of those or something else will meet your needs; if not, then respond accordingly.

**Comment by danhammari** [comment]
Thanks for the feedback. I have gone ahead and extended the`ceLTIc\LTI\Http\HttpMessage` class in my project and added a public `testing` property to it. I also extended the `ceLTIc\LTI\Http\CurlClient` class and added an optional parameter to the `send()` method to receive the `testing` property from the extended `HttpMessage` class. Now I can set the `testing` flag to true in my extended `HttpMessage` class and it sends the `testing` flag to my extended `CurlClient` class `send()` method and that now enables the `curl_setopt` flags to ignore host verification.

**Comment by danhammari** [comment]
I ended up also extending the `ceLTIc\LTI\Http\HttpMessage` class due to how it statically initializes its `self::$httpClient` property. I had to copy its static `getHttpClient()` method to my extended class so it would have the correct scope context and build my extended `CurlClient` class rather than the original `ceLTIc\LTI\Http\CurlClient` class. This seemed the simplest way to resolve the connection between the two classes without having to change anything in the original library.
